### PR TITLE
add / link about_dialog to gui

### LINKED ISSUE #21: Implement "About" menu / popup
Would make the "About" selection functional within the popover menu.

Perhaps something like this? I don't know how to implement it into SamRewritten, however.

0-Screenshot_20191020_233945

### PR DESCRIPTION
**Pull request by telans**
Well I've managed to call the dialog when "About" is clicked, and not destroy it when closed so it can be re-opened. Though do I need to make `stop_about_dialog()` for this?, ~~and is `gtk_widget_set_visible FALSE` what I should be using?~~ I think I'll use `gtk_widget_hide_on_delete`

As for actually using the Screenshot_20191024_095449(Close) button on the dialog, I'm having a little trouble linking a signal-handler to it. Any suggestions? I can add another button next to it with a signal but not the default button as it seems to be a part of the dialog itself. Perhaps using GtkAboutDialog isn't the best idea?

Thanks for the help

Closes #21

### CODE REVIEW

File: glade/main_window.glade [hunk 1]
```diff
@@ -26,6 +26,46 @@ Author: Paul Combaldieu
...
+    <property name="can_focus">False</property>
+    <property name="icon">../assets/icon.png</property>
+    <property name="type_hint">dialog</property>
+    <property name="program_name">SamRewritten</property>
+    <property name="version">A Steam Achievement Manager For Linux</property>
+    <property name="comments" translatable="yes">Contribute to us on GitHub!</property>
+    <property name="website">https://github.com/PaulCombal/SamRewritten</property>
+    <property name="website_label" translatable="yes">Source Code</property>
+    <property name="authors">Paul Combal (PaulCombal)
+William Pierce (wgpierce)</property>
```
Comment (wgpierce) [inline glade/main_window.glade]: feel free to add yourself as a contributor/tester if you like :)

File: glade/main_window.glade [hunk 2]
```diff
@@ -174,12 +215,12 @@ He's pretty confident they won't take much longer to arrive, but is also wrong m
             <property name="visible">True</property>
             <property name="can_focus">True</property>
             <property name="receives_default">True</property>
-            <property name="popover">popovermenu</property>
+            <property name="popover">search_popover</property>
```
Comment (wgpierce) [inline glade/main_window.glade]: yeah these showed up on my glade too, so they're fine.

File: src/gui/MainPickerWindow.h [hunk 3]
```diff
@@ -99,6 +99,10 @@ class MainPickerWindow {
      */
     void switch_to_games_page();
 
+    void on_about_dialog_show();
```
Comment (wgpierce) [inline src/gui/MainPickerWindow.h]: These next few comments are some personal preferences.
All the other functions in MainPickerWindow are verbs, so I suggest show_about_dialog and hide_about_dialog

File: src/gui/gtk_callbacks.h [hunk 4]
```diff
@@ -64,6 +64,12 @@ extern "C"
     void 
     on_main_window_show();
```
Comment (wgpierce) [inline src/gui/gtk_callbacks.h:66]: I suggest on_about_button_clicked and on_about_dialog_close or on_about_dialog_close_button_clicked

**Comment by wgpierce** [comment]
Left some comments above.

> Any suggestions? I can add another button next to it with a signal but not the default button as it seems to be a part of the dialog itself. Perhaps using GtkAboutDialog isn't the best idea?

Looks like dialogs are a little special with their signals. What you want is to connect the response signal to the same handler as the close handler: https://developer.gnome.org/gtk3/stable/GtkDialog.html#GtkDialog-response. You don't even have to determine what signal it is since the about dialog only gives off one signal - close.

Take care of those and I'll accept it!

**Comment by wgpierce** [comment]
Also - update the original commit instead of creating a separate one on top of the old one (git commit --amend or git merge --squash if you already have multiple)

**Comment by telans** [comment]
> Looks like dialogs are a little special with their signals. What you want is to connect the response signal to the same handler as the close handler: https://developer.gnome.org/gtk3/stable/GtkDialog.html#GtkDialog-response. You don't even have to determine what signal it is since the about dialog only gives off one signal - close.

I'm really not getting anywhere with this one, it's flying a little over my head. I think I'll need to see it in action before it clicks. Pushed the other changes though

Thanks for your help.

**Comment by wgpierce** [comment]
GtkAboutDialog inherits properties from GtkDialog. One of those properties is the signal for the "response" from user input. That's where the signal from the button ends up. Check the signals tab in glade and hook the same close function up to that signal.

**Comment by telans** [comment]
Well that was awfully simple, didn't realise that was there. Thanks.

Should all be working now

File: glade/main_window.glade [hunk 5]
```diff
@@ -174,12 +217,12 @@ He's pretty confident they won't take much longer to arrive, but is also wrong m
             <property name="visible">True</property>
             <property name="can_focus">True</property>
             <property name="receives_default">True</property>
```
Comment (wgpierce) [inline glade/main_window.glade]: actually I noticed this is swapping the two buttons, can you revert the changes below this line in this file?

**Comment by wgpierce** [comment]
You got it! Sorry, I left one more comment above. Revert that and I'll accept it.

**Comment by telans** [comment]
Should all be taken care of. Any idea why Glade is changing them around? I'm sure I didn't do it but I may have clicked something wrong

**Comment by wgpierce** [comment]
>  Any idea why Glade is changing them around?

Not sure why :/

Anyway looks good!
